### ISSUE TITLE
Invalid index into object stream given (ArgumentError)

### ISSUE THREAD

**Issue by earthlingworks** [gettalong/hexapdf#246]
Hi Thomas! Hope all is well :) 

We have an issue with some PDFs. Error: `Invalid index into object stream given (ArgumentError)`

Will send the PDF in a separate email.

Script to reproduce:

```require 'hexapdf'
​
path = 'output.pdf'
​
document = HexaPDF::Document.open(ARGV[0])
​
document.write(path, validate: false, optimize: true)
​
document2 = HexaPDF::Document.open(path)```

**Comment by gettalong** [comment]
I can reproduce the problem and I found the bug responsible for this.

**Comment by gettalong** [comment]
Fixed with the new release 0.32.2!

**Comment by earthlingworks** [comment]
Excellent - thank you!
